keps/sig-scheduling/3902-decoupled-taint-manager/README.md

```diff
@@ -0,0 +1,266 @@
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to existing tests to make this code solid enough prior to committing the changes necessary to implement this enhancement.
+
+##### Prerequisite testing updates
+
+Kubernetes already has a good coverage of node `No-Execute` eviction. We will add tests only for the changed code.
+
+#####  Unit tests
+- `pkg/controller/taintmanager`: 2023-05-23 - 0%  
+  - Enable and disable the new `TaintManager`
+  - The new `TaintManager` acts on node taints properly
+- `pkg/controller/apis/config/v1alpha1`: 2023-05-23 - 0% 
+  - Use the split `TaintManager` or the old combined controller
+
+#####  Integration tests
+ No integration tests are required. This feature is better tested with e2e tests.  
+
+##### E2E tests
+
+- Verify passing the existing E2E and conformance tests using the split `TaintManager`.
+- Verify the ability to disable the default `TaintManager`.
+
+### Graduation Criteria
+
+#### Alpha
+
+Since there are no new API changes, we can skip Alpha and go to Beta directly.
+
+#### Beta
+
+* Support `kube-controller-manager` flag `--controllers=-taint-manager` to disable the default taint-manager.
+* Unit and e2e tests completed and passed.
+* Performance and scalability tests to verify there is non-negligible performance degradation in node taint-based eviction.
+* Update documents to reflect the changes.
+
+#### GA
+
+* Fix all reported bugs if any.
+
+### Upgrade / Downgrade Strategy
+
+We’ll need to upgrade `kube-controller-manager` and` node-lifecycle-controller` and add a new controller `taint-manager`. Accordingly, a downgrade will revert to the previous version of `kube-controller-manager` and `node-lifecycle-controller` and remove the new added `taint-manager` . There should be no impact on the other components during upgrades and downgrades since user-exposed behavior should stay the same except a new flag to enable/disable the default `taint-manager`. This change should be clearly communicated to users.
+
+### Version Skew Strategy
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+*This section must be completed when targeting alpha to a release.*
+
+* **How can this feature be enabled / disabled in a live cluster?**
+    * Enabled:  turn on the new feature gate `SeparateTaintManager`
+    * Disabled: turn off the new feature gate `SeparateTaintManager`
+* **Does enabling the feature change any default behavior?**
+    * No, the default node-life-cycle controller and taint-manager behavior will stay the same.
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?**
+    * Yes, we will need to rack back `kube-controller-manager` and `node-lifecycle-controller`. 
+* **What happens if we re-enable the feature if it was previously rolled back?**
+    * Upgrade `kube-controller-manager` and `node-lifecycle-controller` and add the new `taint-manager`.
```

Comment:
This has a similar problem like my other comment.

Reply:
Changed it to  "Use the feature gate `SeparateTaintManager` to enable or disable the feature."